src/order/filter/ennreal.lean

```diff
@@ -0,0 +1,102 @@
+/-
+Copyright (c) 2021 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+
+import data.real.ennreal
+import order.filter.countable_Inter
+import order.liminf_limsup
+
+/-!
+# Order properties of extended non-negative reals
+
+This file compiles filter-related results about `ennreal` (see data/real/ennreal.lean).
+-/
+
+open filter
+open_locale filter
+
+namespace ennreal
+variables {α : Type*} {f : filter α}
+
+lemma eventually_le_limsup [countable_Inter_filter f] (u : α → ennreal) :
+  ∀ᶠ y in f, u y ≤ f.limsup u :=
+begin
+  by_cases hx_top : f.limsup u = ⊤,
+  { simp_rw hx_top,
+    exact eventually_of_forall (λ a, le_top), },
+  have h_forall_le : ∀ᶠ y in f, ∀ n : ℕ, u y < f.limsup u + (1:ennreal)/n,
+  { rw eventually_countable_forall,
+    refine λ n, eventually_lt_of_limsup_lt _,
+    nth_rewrite 0 ←add_zero (f.limsup u),
+    exact (add_lt_add_iff_left (lt_top_iff_ne_top.mpr hx_top)).mpr (by simp), },
+  refine h_forall_le.mono (λ y hy, le_of_forall_epsilon_le (λ r hr_pos hx_top,_)),
+  have hr_ne_zero : (r : ennreal) ≠ 0,
+  { rw [ne.def, coe_eq_zero],
+    exact (ne_of_lt hr_pos).symm, },
+  cases (exists_inv_nat_lt hr_ne_zero) with i hi,
+  refine le_trans (le_of_lt (hy i)) (add_le_add_left (le_of_lt _) (f.limsup u)),
+  rwa [div_def, one_mul],
+end
+
+lemma limsup_eq_zero_iff [ne_bot f] [countable_Inter_filter f] {u : α → ennreal} :
```

Comment:
Isn't the assumption `ne_bot f` not necessary here? If `f = bot`, then I guess the limsup of every function is `0`, and every function is almost everywhere 0, so both sides are satisfied. Same thing in the next two lemmas, both sides are zero when `f = bot`. Same thing in the applications to ess_sup.

Reply:
Thanks. I think that all those unnecessary hypotheses you keep finding in my results are due to me adding them whenever I need them in the "main" proof, and then not taking the time to look back at whether they are really necessary afterwards. I'll make a note to systematically check hypotheses.